packages/odata/src/services/grid-odata.service.ts

```diff
@@ -428,7 +428,16 @@ export class GridOdataService implements BackendService {
           } else if (operator === OperatorType.rangeExclusive || operator === OperatorType.rangeInclusive) {
             // example:: (Name >= 'Bob' and Name <= 'Jane')
             searchBy = this.filterBySearchTermRange(fieldName, operator, searchTerms);
-          } else if ((operator === '' || operator === OperatorType.contains || operator === OperatorType.notContains) &&
+          } else if (this._gridOptions && (this._gridOptions.caseInsensitiveOdataSearch || !this._gridOptions.hasOwnProperty('caseInsensitiveOdataSearch'))) {
```

Comment:
I don't think it necessary to add the flag check in the `if` condition, I would rather you simply use the flag in the code below (untested)

```diff
+ if (this._gridOptions.caseInsensitiveOdataSearch) {
+   searchBy = odataVersion >= 4 ? `contains(tolower(${fieldName}, ${searchValue}))` : `substringof(tolower(${searchValue}, ${fieldName}))`;
+ else {
  searchBy = odataVersion >= 4 ? `contains(${fieldName}, ${searchValue})` : `substringof(${searchValue}, ${fieldName})`;
+ }
```

Reply:
it also looks like the linter failed as well, make sure to have the correct indentation in your code. I use the Editorconfig VSCode extension and also auto-format on save when I run the project. We also need unit tests to cover the change, if you need help let me know, here's 1 of the test for `substringof`, I would expect to see at least 2 new unit tests covering your change for both `contains(tolower(...))` and ``substringof(tolower(...))`

https://github.com/ghiscoding/slickgrid-universal/blob/9826623ee32fd3dd3b5f34f4f86e5d1c3ce02027/packages/odata/src/services/__tests__/grid-odata.service.spec.ts#L1057-L1071